packages/amplify-category-geo/src/service-walkthroughs/deviceLocationTrackingWalkthrough.ts

```diff
@@ -111,3 +114,76 @@ export const deviceLocationTrackerAccessWalkthrough = async (
+ */
+export const updateDeviceLocationTrackerWalkthrough = async (
+  context: $TSContext,
+  parameters: Partial<DeviceLocationTrackingParameters>,
+  resourceToUpdate?: string,
+): Promise<Partial<DeviceLocationTrackingParameters>> => {
+  let resourceName = resourceToUpdate;
+  let updatedParameters = parameters;
+  const trackingResourceNames = await getGeoResources(ServiceName.DeviceLocationTracking);
+  if (trackingResourceNames.length === 0) {
+    printer.error(`No ${trackingServiceFriendlyName} resource to update. Use "amplify add geo" to create a new ${trackingServiceFriendlyName}.`);
+    return updatedParameters;
+  }
+
+  if (resourceName) {
+    if (!trackingResourceNames.includes(resourceName)) {
+      printer.error(`No ${trackingServiceFriendlyName} named ${resourceName} exists in the project.`);
+      return updatedParameters;
+    }
+  } else {
+    resourceName = await prompter.pick<'one', string>(`Select the ${trackingServiceFriendlyName} you want to update`, trackingResourceNames);
+  }
+
+  updatedParameters.name = resourceName;
+  updatedParameters = merge(updatedParameters, await getCurrentTrackingParameters(resourceName));
+
+  // overwrite the parameters based on user input
+
+  const deviceParams = await deviceLocationTrackerAccessWalkthrough(context, updatedParameters);
+  updatedParameters.groupPermissions = deviceParams.groupPermissions;
+  updatedParameters.roleAndGroupPermissionsMap = deviceParams.roleAndGroupPermissionsMap;
+
+  const otherTrackingResources = trackingResourceNames.filter(trackingResourceName => trackingResourceName !== resourceName);
+  // if this is the only device tracker, default cannot be removed
+  if (otherTrackingResources.length > 0) {
+    const isDefault = await prompter.yesOrNo(defaultResourceQuestion(ServiceName.DeviceLocationTracking), updatedParameters.isDefault);
+    // If a device tracker is updated, ask for new default
+    if (updatedParameters.isDefault && !isDefault) {
+      await updateDefaultDeviceTrackerWalkthrough(context, resourceName, otherTrackingResources);
+    }
+    updatedParameters.isDefault = isDefault;
+  } else {
+    updatedParameters.isDefault = true;
+  }
+  return updatedParameters;
+};
+
+/**
+ * Walkthrough to choose a different default device tracker
+ * @param context The Amplify Context object
+ * @param currentDefault The current default device tracker name
+ * @param availableDeviceTrackers The names of available device trackers
+ * @returns name of the new default device tracker chosen
+ */
+export const updateDefaultDeviceTrackerWalkthrough = async (
+  context: $TSContext,
+  currentDefault: string,
+  availableDeviceTrackers?: string[],
+): Promise<string> => {
+  let trackers = availableDeviceTrackers;
+  if (!trackers) {
+    trackers = await getGeoResources(ServiceName.DeviceLocationTracking);
+  }
+  const otherDeviceTrackers = trackers.filter(name => name !== currentDefault);
+  if (otherDeviceTrackers?.length > 0) {
```

Comment:
If there's only one option, prompter.pick will automatically skip the prompt and return the single option